src/commands/ingress/toggleIngressVisibility/ToggleIngressVisibilityStep.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import type { Progress } from "vscode";
+import { IngressConstants } from "../../../constants";
+import { localize } from "../../../utils/localize";
+import type { IngressContext } from "../IngressContext";
+import { IngressUpdateBaseStep } from "../IngressUpdateBaseStep";
+
+export class ToggleIngressVisibilityStep extends IngressUpdateBaseStep<IngressContext> {
+    public priority: number = 300;
+
+    public async execute(context: IngressContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const containerApp = nonNullProp(context, 'containerApp');
+        const ingress = nonNullValueAndProp(containerApp.configuration, 'ingress');
+
+        const warningPrompt = localize('visibilityWarning', 'This will change the ingress visibility from "{0}" to "{1}".', ingress.external ? IngressConstants.external : IngressConstants.internal, !ingress.external ? IngressConstants.external : IngressConstants.internal)
+        await context.ui.showWarningMessage(warningPrompt, { modal: true }, { title: localize('continue', 'Continue') });
+        ingress.external = !ingress.external;
+
+        const working: string = localize('updatingVisibility', 'Updating ingress visibility...');
+        const workCompleted: string = localize('updatedVisibility', 'Updated container app "{0}" ingress visibility to "{1}"', containerApp.name, ingress.external ? IngressConstants.external : IngressConstants.internal);
+
+        await this.updateIngressSettings(context, progress, { ingress, working, workCompleted });
+    }
+
+    public shouldExecute(): boolean {
+        return true;
```

Comment:
I've also noticed other steps where `shouldExecute` just returns true.  I think this is another practice that goes against our typical wizard design patterns. 

Typically, we want to think about the proper conditions for these steps to execute. Again, since this step isn't super reusable,it doesn't matter very much in this case, but the context for properties that the user set should be validated to ensure that there weren't any errors with prompting. I think in general, it will make code less error prone and debugging easier.

A bit of a tangent, but I kind of want to execute steps to log when they don't execute and which values were missing that caused them not to execute for further clarity.

Reply:
Noted, though in this case it wasn't super clear to me if we needed to even check for anything here since it's just a toggle.  This did make me reconsider some of the other sections though... it's perhaps relevant to add a more specific check when we edit the target port.